vector/src/main/java/im/vector/activity/MediaPreviewerActivity.java

```diff
@@ -27,187 +28,126 @@
+    TextView mFileNameView;
 
-    @NotNull
-    @Override
-    public Pair getOtherThemes() {
-        return new Pair(R.style.AppTheme_NoActionBar_Dark, R.style.AppTheme_NoActionBar_Black);
-    }
 
     @Override
     public int getLayoutRes() {
         return R.layout.activity_media_previewer;
     }
 
-    @Override
-    public void onItemPositionChangedListener(int position) {
-        setPreview(mSharedDataItems.get(position));
-    }
-
+    @SuppressLint("ClickableViewAccessibility")
     @Override
     public void initUiAndData() {
-
         if (CommonActivityUtils.shouldRestartApp(this)) {
             Log.d(LOG_TAG, "onCreate : restart the application");
             CommonActivityUtils.restartApp(this);
             return;
         }
-
         if (CommonActivityUtils.isGoingToSplash(this)) {
             Log.d(LOG_TAG, "onCreate : Going to splash screen");
             return;
         }
-
+        mPreviewerVideoView.setOnTouchListener(new View.OnTouchListener() {
+            @Override
+            public boolean onTouch(View v, MotionEvent event) {
+                onVideoPreviewClicked();
+                return false;
+            }
+        });
         configureToolbar();
-
-        String roomTitle = (String) getIntent().getExtras().get(EXTRA_ROOM_TITLE);
-        if (!TextUtils.isEmpty(roomTitle)) {
-            getSupportActionBar().setTitle(roomTitle);
-        }
-
-        setStatusBarColor(findViewById(R.id.status_bar_background),
-                ContextCompat.getColor(this, R.color.transparent_dark));
-
+        final String roomTitle = getIntent().getExtras().getString(EXTRA_ROOM_TITLE);
+        getSupportActionBar().setTitle(roomTitle);
         // Resize web content to prevent scrollbars.
-        mWebPreview.getSettings().setUseWideViewPort(true);
-        mWebPreview.getSettings().setLoadWithOverviewMode(true);
-
-        mSharedDataItems = new ArrayList<>(RoomMediaMessage.listRoomMediaMessages(getIntent(), RoomMediaMessage.class.getClassLoader()));
-
-        if (mSharedDataItems.isEmpty()) {
-            mSharedDataItems.add(new RoomMediaMessage(Uri.parse(getIntent().getStringExtra(EXTRA_CAMERA_PICTURE_URI))));
+        final List<RoomMediaMessage> sharedDataItems = new ArrayList<>(RoomMediaMessage.listRoomMediaMessages(getIntent(), RoomMediaMessage.class.getClassLoader()));
+        if (sharedDataItems.isEmpty()) {
+            sharedDataItems.add(new RoomMediaMessage(Uri.parse(getIntent().getStringExtra(EXTRA_CAMERA_PICTURE_URI))));
+        } else {
+            final LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false);
+            mPreviewerRecyclerView.setLayoutManager(linearLayoutManager);
+            final MediaPreviewAdapter mediaPreviewAdapter = new MediaPreviewAdapter(sharedDataItems, this);
+            mPreviewerRecyclerView.setAdapter(mediaPreviewAdapter);
+            updatePreview(sharedDataItems.get(0));
```

Comment:
It's strange, it's working, but I think this line should be outside the else block, no?

Reply:
Actually it's not working if you select "Use native camera" in the settings. Can you check this case pls?

Reply:
Yes, just saw that too